src/app/containers/PageHandlers/withVariant/getVariantRedirectUrl/index.js

```diff
@@ -0,0 +1,40 @@
+import path from 'ramda/src/path';
+import pathToRegexp from 'path-to-regexp';
+import { getVariant, variantSanitiser } from '#lib/utilities/variantHandler';
+import { getPreferredVariant } from '#app/contexts/UserContext/cookies';
+
+const getVariantRedirectUrl = (props, service, variant) => {
+  const cookieVariant = getPreferredVariant(service);
+  const defaultVariant = getVariant({
+    service,
+    variant: variantSanitiser(variant),
+  });
+  const redirectVariant = cookieVariant || defaultVariant;
```

Comment:
If the cookie is set incorrectly, it shows a white screen after the redirect. We should ensure the cookie is a valid variant before using it.

Reply:
Fixed!